### PR TITLE
Hotfix/metadata migration

### PR DESCRIPTION
**Pull request by stackchain**

### CODE REVIEW

File: src/components/ErrorBoundary/ErrorBoundary.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
+import React, {Component, ErrorInfo, ReactNode} from 'react'
```
Comment (wolverineks) [inline src/components/ErrorBoundary/ErrorBoundary.tsx:1]: use react-error-boundary package instead

File: src/components/ErrorBoundary/ErrorBoundary.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
```
Comment (wolverineks) [inline src/components/ErrorBoundary/ErrorBoundary.stories.tsx]: not needed

File: src/components/ErrorBoundary/ErrorBoundary.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+  render() {
+    if (this.state.hasError) {
+      return (
+        <View
+          style={{
+            flex: 1,
+            justifyContent: 'center',
+            alignItems: 'center',
+            marginHorizontal: 12,
```
Comment (wolverineks) [inline src/components/ErrorBoundary/ErrorBoundary.tsx]: avoid margin when possible

File: src/components/ErrorBoundary/ErrorBoundary.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            paddingTop: 80,
+          }}
+        >
+          <Text>Oops!!! Something went wrong..</Text>
```
Comment (wolverineks) [inline src/components/ErrorBoundary/ErrorBoundary.tsx]: is there a figma for this? if not, there should be

**Review by wolverineks — CHANGES_REQUESTED** [review]

File: legacy/crypto/walletManager.js [hunk 5]
```diff
@@ -128,6 +128,27 @@ class WalletManager {
           checksum = w.checksum
         }
         const isHW = data != null && data.isHW != null ? data.isHW : false
+
+        // migration - missing checksum
+        const metadataUpdate: WalletMeta = {...w}
```
Comment (wolverineks) [inline legacy/crypto/walletManager.js]: can this be pulled out and given a name?
would it make sense to add a test for these 2 migrations?

File: src/components/ErrorBoundary/ErrorBoundary.tsx (hunk 1 above)
Comment (wolverineks) [reply src/components/ErrorBoundary/ErrorBoundary.tsx:1]: https://github.com/bvaughn/react-error-boundary

Comment (stackchain) [reply src/components/ErrorBoundary/ErrorBoundary.tsx:1]: what are the advantages of using this package?

File: src/components/ErrorBoundary/ErrorBoundary.tsx (hunk 4 above)
Comment (stackchain) [reply src/components/ErrorBoundary/ErrorBoundary.tsx]: atm there is not :/

File: legacy/crypto/walletManager.js (hunk 5 above)
Comment (stackchain) [reply legacy/crypto/walletManager.js]: 🤔

File: src/components/ErrorBoundary/ErrorBoundary.tsx (hunk 1 above)
Comment (stackchain) [reply src/components/ErrorBoundary/ErrorBoundary.tsx:1]: I was checking it, is pretty stuffed with things that I don't see we using for now, like a hook to set the error, a throw will do the job, what I liked was the "reset" impl, I will sync with Dima to cover on Figma, perhaps we can add it later.

Comment (wolverineks) [reply src/components/ErrorBoundary/ErrorBoundary.tsx:1]: it's excellent, very popular, well maintained by a Facebook relay developer, fully tested

Comment (stackchain) [reply src/components/ErrorBoundary/ErrorBoundary.tsx:1]: Ticket created to add it, don't want to add any packages in prod, then we can move the current errorboundary as the fallback screen of it.

File: src/appStorage.ts [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+}
+
+function migrateAttribute(attr: keyof WalletMeta, walletMetaUpdated: Partial<WalletMeta>): Promise<void> {
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const id = walletMetaUpdated!.id
+  Logger.warn(`migrateAttribute::${attr} of wallet ${id}`)
+  return storage.write(`/wallet/${id}`, walletMetaUpdated)
+}
+
+function isWalletMeta(walletMeta?: WalletMeta | object | undefined): walletMeta is WalletMeta {
```
Comment (wolverineks) [inline src/appStorage.ts]: `?` implies undefined

Comment (wolverineks) [reply src/appStorage.ts]: when migrating to ts, i think `unknown` might be the best type

File: src/appStorage.ts [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+    'isEasyConfirmationEnabled' in walletMeta 
+      && typeof walletMeta.isEasyConfirmationEnabled === 'boolean' &&
+    'checksum' in walletMeta 
+      && typeof walletMeta.checksum === 'object' &&
+    ('provider' in walletMeta 
+      && typeof walletMeta.provider === 'string'
+      || !('provider' in walletMeta)) &&
+    'walletImplementationId' in walletMeta 
+      && typeof walletMeta.walletImplementationId === 'string' 
+      && Object.values(WALLET_IMPLEMENTATION_REGISTRY).indexOf(walletMeta.walletImplementationId) > -1 &&
```
Comment (wolverineks) [inline src/appStorage.ts]: .includes

**Review by wolverineks — APPROVED** [review]

File: src/appStorage.ts (hunk 7 above)
Comment (stackchain) [reply src/appStorage.ts]: 👍  copy and paste from the old validator

**Comment by stackchain** [comment]
Release: https://github.com/Emurgo/yoroi-mobile/releases/tag/4.7.12

**Pull request merged**